src/App/App.js

```diff
@@ -25,16 +28,33 @@ class App extends Component {
   render() {
     return (
       <div className='App'>
-        <Header />
-        <main>
-          {this.state.error && 
-          <h3 className='error-msg'>{this.state.error}</h3>
-          }
-          <Movies movies={this.state.movies} />
-        </main>
+        <Header showLogin={this.showLogin} />
+        {this.state.view === 'homepage' && 
+          <>
+            <main>
+              {this.state.error && 
+              <h3 className='error-msg'>{this.state.error}</h3>
+              }
+              <Movies movies={this.state.movies} />
+            </main>
+          </> 
+        }
+
+        {this.state.view === 'login' && 
+          <Login updateUserId={this.updateUserId}/>
+        }
       </div>
     )
   }
+
+  showLogin = () => {
+    this.setState({view: 'login'})
+  }
```

Comment:
Maybe we can make this dynamic as it should make it easier to switch between different views? The child components can pass in the parameter i.e. 'login' for this one.
`changeView = (view) => {
      this.setState({view: view})`